window.c

```diff
@@ -486,10 +488,20 @@ void win_draw_rect(win_t *win, int x, int y, int w, int h, bool fill, int lw,
 		XDrawRectangle(win->env.dpy, win->buf.pm, gc, x, y, w, h);
 }
 
-void win_set_title(win_t *win, const char *title)
+void win_set_title(win_t *win, const char *path)
 {
-	XStoreName(win->env.dpy, win->xwin, title);
-	XSetIconName(win->env.dpy, win->xwin, title);
+	const unsigned int title_max = strlen(path) + strlen(options->title_prefix) + 1;
+	char title[title_max];
+	const char *basename = strrchr(path, '/') + 1;
```

Comment:
If path doesn't contain a "/", won't this segfault. `strrchr(path, '/')` would return null. I am assuming path is the path of the file relative to the current directly and not always an absolute path.

Reply:
The path was acquired via `realpath` in `check_add_files` and checks for NULL was made. So it's practically impossible for `strrchr(path, '/')` to return NULL.

Reply:
On my system, `path` is the full path. I do not know if this is system-dependent.

Reply:
`realpath` will always return full path or NULL on failure. We already error out of failure in `check_add_files`, so this is perfectly fine, although a bit hacky.

Reply:
welp, tested and tried real good to make it segfault but i couldn't so unless there's something else we can proceed with the approval

Reply:
Ok cool. Well not cool. I think it is interesting they we use the full path and not the relative one. I feel like that would add a lot of extra noise as probably only a subset of the path is useful info. Well I don't have a immediate use case for the relative path, but the fact the absolute path should probably be highlighted in the docs as that isn't what I was expecting